### ISSUE TITLE
Crash with opened repository in 2.7.0

### ISSUE THREAD

**Issue by raandoom** [gitahead/gitahead#646]
Version 2.7.0 can work with clean start, but can't start after opening an existing working copy, closing the application and trying to start again.

Looks like it saves recent repository in settings (Windows registry) and trying to select current item (current repo) and crash somewhere here.

Crash dump says that segfault appears after signal `selectionChanged()`, maybe somewhere in a connected slot. 

OS: Windows 10

**Comment by raandoom** [comment]
Here it is (short version, full version in attached file):
```
Qt6Gui!QWindow::devicePixelRatio
GitAhead!git_midx_writer_dump+0xb532d
GitAhead!git_midx_writer_dump+0xb4b54
GitAhead!git_midx_writer_dump+0x8e146
GitAhead!git_midx_writer_dump+0x7c025
GitAhead!git_midx_writer_dump+0x2c3db
Qt6Core!QObject::qt_static_metacall+0x17ab
Qt6Core!QMetaObject::activate+0x84
GitAhead!git_midx_writer_dump+0x42879
GitAhead!git_midx_writer_dump+0x68b0b
Qt6Core!QObject::qt_static_metacall+0x17ab
Qt6Core!QMetaObject::activate+0x84
Qt6Core!QItemSelectionModel::emitSelectionChanged+0xa6b
Qt6Core!QItemSelectionModel::select+0x31c
GitAhead!git_midx_writer_dump+0x6d2ec
GitAhead!git_midx_writer_dump+0x6d20a
GitAhead!git_midx_writer_dump+0x6075c
Qt6Core!QObject::qt_static_metacall+0x17ab
Qt6Core!QMetaObject::activate+0x84
GitAhead!git_midx_writer_dump+0x1ef9ca
GitAhead!git_midx_writer_dump+0x3cdb7
GitAhead!git_midx_writer_dump+0x17aff
GitAhead!git_midx_writer_dump+0x1700a
GitAhead!git_midx_writer_dump+0x1907f
GitAhead!git_midx_writer_dump+0x1a2cf
GitAhead!git_midx_writer_dump+0xe248
GitAhead!git_midx_writer_dump+0xbb46
GitAhead!git_midx_writer_dump+0x27076d
GitAhead!git_midx_writer_dump+0x26e21a
kernel32!BaseThreadInitThunk+0x14
ntdll!RtlUserThreadStart+0x21
```

Another information: it is enough to clean up `HKEY_CURRENT_USER\SOFTWARE\gitahead.com\GitAhead\windows\XXXX\path` variable and GitAhead is succesfully started. But if I keep repo opened (`path` variable is not empty) GitAhead cant start.

[gitahead.stacktrace.txt](https://github.com/gitahead/gitahead/files/13665102/gitahead.stacktrace.txt)

**Comment by ordonezt** [comment]
I think the same thing may be happening to me, I had several repositories open and after updating to version 2.7.0 the application no longer starts.
I reinstalled version 2.6.3 and everything works fine.
I'm on Windows 10 Pro 64 bits.

**Comment by hackhaslam** [comment]
It seems to be sensitive to the repository being restored. For example, I can reproduce with the git repository but not with the gitahead repository. I suspect that it's related to repository size. In any case this is fixed by e5b4cef79ded33179608ffb66ea3dddd0ee393af. I'll release another build very soon.

**Comment by hackhaslam** [comment]
This fix is now released in v2.7.1.

**Comment by raandoom** [comment]
Thanks a lot, version 2.7.1 is working :)
